src/sqlfluff/testing/rules.py

```diff
@@ -79,7 +81,11 @@ def assert_rule_fail_in_sql(code, sql, configs=None, line_numbers=None):
                 )
             )
     # The query should already have been fixed if possible so just return the raw.
-    return linted.tree.raw
+    if linted.num_violations(fixable=True) > 0:
+        fixed, _ = linted.fix_string()
```

Comment:
Fixes a bug where YAML rule tests (e.g. L003.yml) were not translating fixed code back to the templated version.